### PR TITLE
document getEntityData

### PR DESCRIPTION
**Pull request by punkle**

### CODE REVIEW

File: content/docs/tech-insights/define-custom-data-sources/index.md [hunk 1]
```diff
@@ -122,6 +122,14 @@ If you want to parse YAML content from plain strings, for example from an API En
 * JSONata query: `$parseYaml(spec.definition).openapi`
 * Type: String
 
+**$getEntityData**
+
+If you want to parse request entity information in the JSONata context you can use the `$getEntityData` function. An example query could be like the following:
+* Data Provider Type: Entity Definition
+* Fact Name: OpenAPI Spec Version
```
Comment (Xantier) [inline content/docs/tech-insights/define-custom-data-sources/index.md:129]:
```suggestion
* Fact Name: First Target Ref of relations
```

File: content/docs/tech-insights/define-custom-data-sources/index.md [hunk 2]
```diff
@@ -122,6 +122,14 @@ If you want to parse YAML content from plain strings, for example from an API En
...
+* JSONata query: `$getEntityData('relations')[0].targetRef`
+* Type: JSON Object
```
Comment (Xantier) [inline content/docs/tech-insights/define-custom-data-sources/index.md:131]:
```suggestion
* Type: string
```

Comment (punkle) [reply content/docs/tech-insights/define-custom-data-sources/index.md:131]: It returns string, array, object, basically whatever path you use.

File: content/docs/tech-insights/define-custom-data-sources/index.md [hunk 3]
```diff
@@ -122,6 +122,14 @@ If you want to parse YAML content from plain strings, for example from an API En
 * JSONata query: `$parseYaml(spec.definition).openapi`
 * Type: String
 
+**$getEntityData**
+
+If you want to parse request entity information in the JSONata context you can use the `$getEntityData` function. An example query could be like the following:
+* Data Provider Type: Entity Definition
```
[the remark this answers is not in the document]
Comment (Xantier) [reply content/docs/tech-insights/define-custom-data-sources/index.md]: it's these: 
image


Supported types

[the remark this answers is not in the document]
Comment (punkle) [reply content/docs/tech-insights/define-custom-data-sources/index.md]: Oh right.. then its http data source that I added it to.

File: content/docs/tech-insights/define-custom-data-sources/index.md (hunk 2 above)
Comment (Xantier) [reply content/docs/tech-insights/define-custom-data-sources/index.md:131]: yup, makes sense

File: content/docs/tech-insights/define-custom-data-sources/index.md (hunk 1 above)
Comment (punkle) [reply content/docs/tech-insights/define-custom-data-sources/index.md:129]: What does fact name mean?

Comment (Xantier) [reply content/docs/tech-insights/define-custom-data-sources/index.md:129]: it depends what kind of example you want to display within this block. Judging from the query below, it is not getting OpenAPI spec version

Comment (punkle) [reply content/docs/tech-insights/define-custom-data-sources/index.md:129]: Ah I see what you mean.

Comment (punkle) [reply content/docs/tech-insights/define-custom-data-sources/index.md]: How about that.

**Review by Xantier — APPROVED** [review]

**Pull request merged**